src/providers/cachedProvider.ts

```diff
@@ -78,16 +78,11 @@ export class CacheProvider extends RateLimitedProvider {
   }
 
   private buildRedisKey(method: string, params: Array<unknown>) {
-    // Only handles eth_getLogs and eth_call right now.
     switch (method) {
       case "eth_getBlockByNumber":
-        return this.getBlockByNumberPrefix + JSON.stringify(params);
-      case "eth_getLogs":
-        return this.getLogsCachePrefix + JSON.stringify(params);
-      case "eth_call":
-        return this.callCachePrefix + JSON.stringify(params);
+        return `${this.cachePrefix}:getBlockByNumber,` + JSON.stringify(params);
       default:
-        throw new Error(`CacheProvider::buildRedisKey: invalid JSON-RPC method ${method}`);
+        return `${this.cachePrefix}:${method},` + JSON.stringify(params);
```

Comment:
@mrice32 can confirm, but I think the existing implementation might have been an optimisation to avoid lots of repeated string operations in the hot path.

Reply:
reverted